plugin/kprom/config.go

```diff
@@ -64,10 +67,18 @@ type RegistererGatherer interface {
 // Registry sets the registerer and gatherer to add metrics to, rather than a
 // new registry. Use this option if you want to configure both Gatherer and
 // Registerer with the same object.
+// Additionally, passing a `nil`  RegistererGatherer allows registering an external registry,
+// exposing kprom as a custom prometheus collector, this option is mutually
+// exclusive with GoCollectors
 func Registry(rg RegistererGatherer) Opt {
 	return opt{func(c *cfg) {
-		c.reg = rg
-		c.gatherer = rg
+		if rg == nil || (reflect.ValueOf(rg).Kind() == reflect.Ptr && reflect.ValueOf(rg).IsNil()) {
```

Comment:
The latter half of this conditional is a bit paranoid -- I've never seen this -- but it doesn't seem to hurt, so ok 👍